chsdi/views/stationboard.py

```diff
@@ -39,68 +37,13 @@ def __init__(self, request):
 
     @view_config(request_method='GET')
     def get_departures(self):
-        current_date = datetime.datetime.now(timezone('Europe/Berlin'))
-        filters = [OevDepartures.stop == self.id, OevDepartures.time > current_date]
-        if self.destination != 'all':
-            filters.append(OevDepartures.destination == self.destination)
-
-        def serialize(t):
-            return t.strftime('%H:%M')
-
-        def serializeDate(t):
-            return t.strftime('%d/%m/%Y %H:%M')
-
-        query = self.request.db.query(OevDepartures).filter(and_(*filters)).order_by(
-            OevDepartures.time
-        ).limit(self.limit)
-
         try:
-            # Not all the stations are in the table Haltestelle
-            results = [{
-                'id': q.stop,
-                'time': serialize(q.time),
-                'label': q.label,
-                'via': q.via,
-                'currentDate': serializeDate(current_date),
-                'departureDate': serializeDate(q.time),
-                'destinationName': q.haltestelle.name,
-                'destinationId': q.haltestelle.id
-            } for q in query if hasattr(q.haltestelle, 'id')]
+            results = self.ot_api.get_departures(self.id, self.limit)
+            if len(results) == 0:
+                results = None
```

Comment:
Test ConnectionError (requets module), if response is not equal to 200 or 300+ (quota exceeded etc..), Exception
and raise http error codes accrodingly

Reply:
thanks